src/framework/theme/components/chat/custom-message.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable, TemplateRef } from '@angular/core';
+
+/**
+ * Service for self register custom templates
+ */
+@Injectable()
+export class NbCustomMessageService {
+
+  protected readonly customMessages = new Map<string, TemplateRef<any>>();
+
+  getMessageTemplate(type: string): TemplateRef<any> | undefined {
+    return this.customMessages.get(type);
+  }
+
+  setMessageTemplate(type: string, template: TemplateRef<any>): void {
```

Comment:
Maybe change to register?
```suggestion
  registerMessageTemplate(type: string, template: TemplateRef<any>): void {
```